pkg/plugin/plugin.go

```diff
@@ -119,15 +123,16 @@ func (p *plugin) CreateOrUpdate(ctx context.Context, cs *api.OpenShiftManagedClu
 		log.Info("starting deploy")
 		err = p.clusterUpgrader.Deploy(ctx, cs, azuretemplate, deployFn)
 	}
-	if err != nil {
+	if err != (*api.PluginError)(nil) {
 		return err
 	}
-
-	// Wait for infrastructure services to be healthy
 	err = p.clusterUpgrader.WaitForInfraServices(ctx, cs)
-	if err != nil {
+	if err != (*api.PluginError)(nil) {
```

Comment:
Are these type casts necessary if the clusterUpgrader functions return errors instead of PluginErrors? Either way the plugin caller needs to cast `error` to `PluginError` so this seems unnecessary.

Reply:
See also https://golang.org/doc/faq#nil_error

Reply:
OK, agreed. I'll always return error instead of PluginError